### PR TITLE
CRNS-141: typescripting MessageInput component and fixing some other issues

### PR DESCRIPTION
**Pull request by virdesai**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: src/components/MessageInput/ActionSheetAttachment.tsx [hunk 1]
```diff
@@ -37,13 +46,34 @@ const ActionSheetTitleText = styled.Text`
   ${({ theme }) => theme.messageInput.actionSheet.titleText.css};
 `;
 
+export type ActionSheetStyles = {
```
Comment (nhannah) [inline src/components/MessageInput/ActionSheetAttachment.tsx:49]: another reason to drop styled components....how as a user do I know what the sdk expects when there is a mix!

File: src/components/MessageInput/hooks/useMessageDetailsForState.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+  const [numberOfUploads, setNumberOfUploads] = useState(0);
+  const [text, setText] = useState(initialValue || '');
+
+  useEffect(() => {
+    if (message && !isEditingBoolean<At, Ch, Co, Ev, Me, Re, Us>(message)) {
+      setText(message?.text || '');
+      const newFileUploads = [];
+      const newImageUploads = [];
+
+      const attachments = message.attachments as Attachment<At>[];
```
Comment (nhannah) [inline src/components/MessageInput/hooks/useMessageDetailsForState.ts:77]: is this another type without the cast?

**Review by nhannah — APPROVED** [review]

File: src/components/MessageInput/ActionSheetAttachment.tsx (hunk 1 above)
Comment (virdesai) [reply src/components/MessageInput/ActionSheetAttachment.tsx:49]: agreed

File: src/components/MessageInput/ImageUploadPreview.tsx [hunk 3]
```diff
@@ -71,7 +112,7 @@ const ImageUploadPreview = ({ imageUploads, removeImage, retryUpload }) => {
         <UploadProgressIndicator
           action={() => {
             if (retryUpload) {
-              retryUpload(item.id);
+              retryUpload({ newImage: item });
```
[the remark this answers is not in the document]
Comment (virdesai) [reply src/components/MessageInput/ImageUploadPreview.tsx:115]: don't think so, this was one of those that idk how it was working before when typescript started showing warnings. but def a breaking change for the props sent into this function now

File: src/components/MessageInput/hooks/useMessageDetailsForState.ts (hunk 2 above)
Comment (virdesai) [reply src/components/MessageInput/hooks/useMessageDetailsForState.ts:77]: nah this is just to get around Immutable so I can use the for loop without a babel error being thrown (which is was before)

**Pull request merged**